libafl/src/events/mod.rs

```diff
@@ -401,7 +392,7 @@ where
 }
 
 /// [`EventFirer`] fires an event.
-pub trait EventFirer: UsesState {
+pub trait EventFirer<I, S> {
```

Comment:
Back to the roots :D 

Will we ever need where State Input and actual Input don't agree?

Reply:
Won't that mean we have to manually specify an input type generic in many situations?

Reply:
Potentially yes. I doubt that people will work with event firers directly often enough for this to affect usability.